libs/langchain/langchain/document_loaders/parsers/pdf_docai.py

```diff
@@ -0,0 +1,273 @@
+"""Module contains a PDF parser based on DocAI from Google Cloud.
+
+You need to install two libraries to use this parser:
+pip install google-cloud-documentai
+pip install google-cloud-documentai-toolbox
+"""
+import logging
+import time
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Iterator, List, Optional, Sequence
+
+from langchain.docstore.document import Document
+from langchain.document_loaders.base import BaseBlobParser
+from langchain.document_loaders.blob_loaders import Blob
+from langchain.utils.iter import batch_iterate
+
+if TYPE_CHECKING:
+    from google.api_core.operation import Operation
+    from google.cloud.documentai import DocumentProcessorServiceClient
+
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class DocAIParsingResults:
+    """A dataclass to store DocAI parsing results."""
+
+    source_path: str
+    parsed_path: str
+
+
+class DocAIPdfLoader(BaseBlobParser):
+    def __init__(
+        self,
+        *,
+        location: str = "us",
+        processor_name: Optional[str] = None,
+    ):
+        """Initializes the parser.
+
+        Args:
+            location: a GCP location where a DOcAI parser is located
+            processor_name: name of a processor
+        """
+        from google.api_core.client_options import ClientOptions
+
+        self._options = ClientOptions(
+            api_endpoint=f"{location}-documentai.googleapis.com"
+        )
+        self._processor_name = processor_name
+        self._client: Optional["DocumentProcessorServiceClient"] = None
+
+    def lazy_parse(self, blob: Blob, **kwargs: Any) -> Iterator[Document]:
```

Comment:
methods should be fully partialied -- all config comes in through the init

```suggestion
    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
```